engine/src/windowing/window.cpp

```diff
@@ -132,6 +133,14 @@ namespace gen
 		log.error("GLFW Error: {} - {}", error, description);
 	}
 
+	void Window::callback_window_size(GLFWwindow * window, int width, int height)
+	{
+		auto * const self = static_cast<Window *>(glfwGetWindowUserPointer(window));
```

Comment:
`Window` should probably have a (private) static member function that does this and returns `Window&`. It should also assert that `self` isn't `nullptr`.

Reply:
Could you elaborate what you mean by this comment further?

Reply:
I mean that other callbacks will also need to do this pointer casting smelly stuff, so it's best to encapsulate it in one place, and return a safe reference. In case somehow the user data pointer gets reset, the `assert(self != nullptr)` part will terminate the program.

```cpp
static Window& getWindow(GLFWwindow* glfw_window) {
    auto * const self = static_cast<Window *>(glfwGetWindowUserPointer(window));
    assert(self != nullptr);
    return *self;
}
```